File: typescript/src/_49_group_anagrams.ts
/*
 * @lc app=leetcode id=49 lang=typescript
 *
 * [49] Group Anagrams
 */

// @lc code=start
function groupAnagrams(strs: string[]): string[][] {
    const groups = new Map<string, string[]>();
    for (const s of strs) {
        const key = s.split('').sort().join('');
        if (!groups.has(key)) groups.set(key, []);
        groups.get(key)!.push(s);
    }
    return [...groups.values()];
};
// @lc code=end

File: kotlin/src/_49_group_anagrams.kt
/*
 * @lc app=leetcode id=49 lang=kotlin
 *
 * [49] Group Anagrams
 */

// @lc code=start
class Solution {
    fun groupAnagrams(strs: Array<String>): List<List<String>> {
        val groups = mutableMapOf<String, MutableList<String>>()
        for (s in strs) {
            val key = String(s.toCharArray().also { it.sort() })
            groups.getOrPut(key) { mutableListOf() }.add(s)
        }
        return groups.values.toList()
    }
}
// @lc code=end
